File: ros/pb_msgs_example/src/talker.cpp
#include "ros/ros.h"
#include "chatter.pb.h"

#include <sstream>

int main(int argc, char** argv)
{
  ros::init(argc, argv, "talker");
  ros::NodeHandle n;
  ros::Publisher pb_chatter_pub = n.advertise<pb_msgs::ShortMessage>("pb_chatter", 1000);
  ros::Publisher counter_pub = n.advertise<pb_msgs::Counter>("counter", 1000);
  
  ros::Rate loop_rate(10);

  int count = 0;
  while (ros::ok())
  {
    pb_msgs::ShortMessage pb_msg;
    ros::Time now = ros::Time::now();
    pb_msg.mutable_stamp()->set_sec(now.sec);
    pb_msg.mutable_stamp()->set_nsec(now.nsec);
    std::stringstream ss;
    ss << "Hello world " << count;
    pb_msg.set_content(ss.str());

    ROS_INFO("%s", pb_msg.content().c_str());
    pb_chatter_pub.publish(pb_msg);

    pb_msgs::Counter counter;
    counter.set_count(count);
    counter_pub.publish(counter);

    ros::spinOnce();

    loop_rate.sleep();
    ++count;
  }

  return 0;
}

File: ros/pb_msgs_example/scripts/talker.py
#!/usr/bin/env python

import rospy
import pb_msgs.msg

def talker():
    pub = rospy.Publisher('pb_chatter', pb_msgs.msg.ShortMessage, queue_size = 10)
    rospy.init_node('talker', anonymous = True)
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        msg = pb_msgs.msg.ShortMessage()
        now = rospy.get_rostime()
        msg.stamp.sec = now.secs
        msg.stamp.nsec = now.nsecs
        msg.content = "Hello world!"
        pub.publish(msg)
        rate.sleep()
        
if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass